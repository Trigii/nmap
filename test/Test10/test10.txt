Authors: Javier Sande Ríos; Tristán Vaquero Povedano

Description: Scan host with a valid and with blacklisted certificate fingerprint and not valid range.

Input:
	* list: blacklist.csv
	* dns-list: dns.csv

Output:
	* Information about the certificate
	* Self-signed: True
	* Verified: True
	* Validity: True
	* Valid names: False
	* Blacklisted: True (Blacklisted subject)
	* DNS resolved: True
	* DNS: IP NOT in valid range

Command: $ nmap -Pn -p443 10.0.3.149 --script untrustedX509certs.nse --script-args list=blacklist.csv dns-list=dns.csv -v -oN test10.out

