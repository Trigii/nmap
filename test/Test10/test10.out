# Nmap 7.91 scan initiated Mon Nov 13 12:12:20 2023 as: nmap -Pn -p443 --script untrustedX509certs.nse --script-args list=blacklist.csv -v -oN test10.out 10.0.3.149 dns-file=dns.csv
Failed to resolve "dns-file=dns.csv".
Failed to resolve "dns-file=dns.csv".
Nmap scan report for 10.0.3.149
Host is up (0.00041s latency).

PORT    STATE SERVICE
443/tcp open  https
| untrustedX509certs: Subject: commonName=10.0.3.149/organizationName=UC3M/stateOrProvinceName=MADRID/countryName=ES/localityName=GETAFE/organizationalUnitName=UC3M Cyber
| Subject Alternative Name: DNS:10.0.3.149, DNS:uc3m.es
| Issuer: commonName=10.0.3.149/organizationName=UC3M/stateOrProvinceName=MADRID/countryName=ES/localityName=GETAFE/organizationalUnitName=UC3M Cyber
| Public Key type: rsa
| Public Key bits: 2048
| Signature Algorithm: sha256WithRSAEncryption
| Not valid before: 2023-11-13T16:44:13
| Not valid after:  2024-11-12T16:44:13
| MD5:   2d44 0756 2a95 7eaa 8673 fad1 ebe0 9697
| SHA-1: 1940 7ae9 07db f321 f0e9 84f3 6ada 9ec3 9b69 b752
| -----BEGIN CERTIFICATE-----
| MIID0TCCArmgAwIBAgIUUPCewRBp2ebBiXQ//KcfFBmre5AwDQYJKoZIhvcNAQEL
| BQAwaDELMAkGA1UEBhMCRVMxDzANBgNVBAgMBk1BRFJJRDEPMA0GA1UEBwwGR0VU
| QUZFMQ0wCwYDVQQKDARVQzNNMRMwEQYDVQQLDApVQzNNIEN5YmVyMRMwEQYDVQQD
| DAoxMC4wLjMuMTQ5MB4XDTIzMTExMzE2NDQxM1oXDTI0MTExMjE2NDQxM1owaDEL
| MAkGA1UEBhMCRVMxDzANBgNVBAgMBk1BRFJJRDEPMA0GA1UEBwwGR0VUQUZFMQ0w
| CwYDVQQKDARVQzNNMRMwEQYDVQQLDApVQzNNIEN5YmVyMRMwEQYDVQQDDAoxMC4w
| LjMuMTQ5MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAo6vOL6wYbe1Y
| 8uCkc6DMmRX90N8VLXjkOCu//U1et5GPo0GAT5ZwdsKD/FTdf9h+fb+ZqflQeWDW
| p5vSuWj76j9gD/2mh/z2eM1h7JiS4H1XMBcPi71Y6k7XsQbBb1cczUZLGgN2Ks8X
| 5dLEKO3oWk+EI34QAGrKkzo/Ev4UIsZOGwac4bfy5/QFAaoYFcT1Y21ed6z7h0yN
| xMWUM1GxSWSNhG0sPC1WZPUNawi1wUbIJ0Gx29XBw9k6F0HegN2xYj4QsxLIeaLX
| GMui0GXG7wx7VYqJMiMs6cpSn51Lj+XiYCadYPT70KRfM5MLBnWeqQZKh9TuR8Rq
| Lfo+8LpqNwIDAQABo3MwcTAeBgNVHREEFzAVggoxMC4wLjMuMTQ5ggd1YzNtLmVz
| MB0GA1UdDgQWBBScPePCZU0iP/y0AYDh1WEH3B9TNDAfBgNVHSMEGDAWgBScPePC
| ZU0iP/y0AYDh1WEH3B9TNDAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUA
| A4IBAQAb7F122SD/mTcxK95jyNaXH9FZzOrbTziS6WnmOrHyHXfShTfnXkRCIjs1
| bAGuyAXcODvF7/C54zgFMh810HazItBH/5oIeT8qoSu0WS2tlH7sGNlRuZrPv4ko
| jU4U55QF/HYL2sX39JCqlX5gkcPppGk5AscjbDt4uaXhTFl/2MPwm9gei+EizAzI
| lGfVveC7EwPHz3CtzPI7aFQXq9FLkVNhBrYsNttkUeKGV5zwfFcrKSEg5xgQhb7r
| 03+tRmTdi8qEbnBAWnPU/L/IAp1inghwgdHRgbKr+JXqWJH4kx1EBcv2oCLLZH/J
| W7qTbRqIqf7cTrWpUWzoAws8IwD6
| -----END CERTIFICATE-----
| 
| --------------------------------
| CERTIFICATE SECURITY REPORT
| --------------------------------
| Self-signed: True
| Certificate verified
| Validity: Certificate is VALID
| Certificate Subject Alternative Names contains hostname
| Blacklisted: Certificate Subject is blacklisted: on 13/11/2023 with severity high
| DNS resolved: True
|_DNS: server ip is NOT IN A VALID RANGE

Failed to resolve "dns-file=dns.csv".
Read data files from: /usr/bin/../share/nmap
# Nmap done at Mon Nov 13 12:12:33 2023 -- 1 IP address (1 host up) scanned in 13.29 seconds
